api/src/main/java/io/grpc/NameResolverRegistry.java

```diff
@@ -81,16 +82,16 @@ public synchronized void deregister(NameResolverProvider provider) {
   }
 
   private synchronized void refreshProviders() {
-    List<NameResolverProvider> providers = new ArrayList<>(allProviders);
-    // Sort descending based on priority.
-    // sort() must be stable, as we prefer first-registered providers
-    Collections.sort(providers, Collections.reverseOrder(new Comparator<NameResolverProvider>() {
-      @Override
-      public int compare(NameResolverProvider o1, NameResolverProvider o2) {
-        return o1.priority() - o2.priority();
+    LinkedHashMap<String, NameResolverProvider> refreshedProviders = new LinkedHashMap<>();
+    // We prefer first-registered providers
+    for (NameResolverProvider provider : allProviders) {
+      String scheme = provider.getDefaultScheme();
```

Comment:
This method doesn't have the correct semantics; I think we need a new API for this, like `getScheme()`. But we can make the default implementation of `getScheme()` delegate to `getDefaultScheme()`.